exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +83,531 @@ type BTCMarkets struct {
+// GetMultipleOrderbooks gets orderbooks
+func (b *BTCMarkets) GetMultipleOrderbooks(marketIDs []string) ([]Orderbook, error) {
+	var orderbooks []Orderbook
+	var temp []tempOrderbook
+	var tempOB Orderbook
+	params := url.Values{}
+	for x := range marketIDs {
+		params.Add("marketId", marketIDs[x])
+	}
+	err := b.SendHTTPRequest(btcMarketsUnauthPath+btcMarketsMultipleOrderbooks+params.Encode(),
+		&temp)
 	if err != nil {
-		return nil, err
+		return orderbooks, err
+	}
+	for i := range temp {
+		var price, volume float64
+		tempOB.MarketID = temp[i].MarketID
+		tempOB.SnapshotID = temp[i].SnapshotID
+		for a := range temp[i].Asks {
+			volume, err = strconv.ParseFloat(temp[i].Asks[a][1], 64)
+			if err != nil {
+				return orderbooks, err
+			}
+			price, err = strconv.ParseFloat(temp[i].Asks[a][0], 64)
+			if err != nil {
+				return orderbooks, err
+			}
+			tempOB.Asks = append(tempOB.Asks, OBData{Price: price, Volume: volume})
+		}
+		for y := range temp[i].Bids {
+			volume, err = strconv.ParseFloat(temp[i].Bids[y][1], 64)
+			if err != nil {
+				return orderbooks, err
+			}
+			price, err = strconv.ParseFloat(temp[i].Bids[y][0], 64)
+			if err != nil {
+				return orderbooks, err
+			}
+			tempOB.Bids = append(tempOB.Bids, OBData{Price: price, Volume: volume})
+		}
+		orderbooks = append(orderbooks, tempOB)
 	}
+	return orderbooks, nil
+}
 
-	if !resp.Success {
-		return nil, errors.New(resp.ErrorMessage)
-	}
+// GetServerTime gets time from btcmarkets
+func (b *BTCMarkets) GetServerTime() (time.Time, error) {
+	var resp TimeResp
+	return resp.Time, b.SendHTTPRequest(btcMarketsAPIURL+btcMarketsAPIVersion+btcMarketsGetTime,
+		&resp)
+}
 
-	for i := range resp.Orders {
-		resp.Orders[i].Price /= common.SatoshisPerBTC
-		resp.Orders[i].OpenVolume /= common.SatoshisPerBTC
-		resp.Orders[i].Volume /= common.SatoshisPerBTC
+// GetAccountBalance returns the full account balance
+func (b *BTCMarkets) GetAccountBalance() ([]AccountData, error) {
+	var resp []AccountData
+	return resp,
+		b.SendAuthenticatedRequest(http.MethodGet,
+			btcMarketsAccountBalance,
+			nil,
+			&resp)
+}
 
-		for j := range resp.Orders[i].Trades {
-			resp.Orders[i].Trades[j].Fee /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[j].Price /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[j].Volume /= common.SatoshisPerBTC
-		}
+// GetTradingFees returns trading fees for all pairs based on trading activity
+func (b *BTCMarkets) GetTradingFees() (TradingFeeResponse, error) {
+	var resp TradingFeeResponse
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		btcMarketsTradingFees,
+		nil,
+		&resp)
+}
+
+// GetTradeHistory returns trade history
+func (b *BTCMarkets) GetTradeHistory(marketID, orderID string, before, after, limit int64) ([]TradeHistoryData, error) {
+	var resp []TradeHistoryData
+	params := url.Values{}
+	if marketID != "" {
+		params.Set("marketId", marketID)
+	}
+	if orderID != "" {
+		params.Set("orderId", orderID)
+	}
+	if before >= 0 {
```

Comment:
Please add a check to ensure that only before or after can be set and before should be > 0